src/libs/cursor.rs

```diff
@@ -382,6 +393,13 @@ where
         self.inc_bytes(1).map_err(WriteError::from)
     }
 
+    fn write_i8(&mut self, data: impl Into<i8>) -> Result<(), WriteError> {
+        let data = data.into().to_le_bytes();
+        let pos = self.byte_pos as usize;
+        self.as_mut()[pos] = data[0];
```

Comment:
Just

```suggestion
        let pos = self.byte_pos as usize;
        self.as_mut()[pos] = data.into();
```

Reply:
With my understanding we cannot do that because `data` is `impl Into<i8>` while self is `AsMut<[u8]>`

Reply:
Ok, agree with your version then